wordpress-mobile/WordPress-Android
# Use `CrashLogging` from `Tracks`

Fixes #14513 

To test:

1. Checkout this PR
2. Apply Git's Patch from the bottom of the description. It basically enables crash reports for debug builds and sends an issue to Sentry when `MySiteFragment` is created.
3. Perform assertions: open some different issue on Sentry and the new one side by side. Compare them, make sure they have the same information. Especially check that

- [ ] user informations are applied
- [ ] breadcrumbs are reported (both app's lifecycle and app's logs)
- [ ] release name is applied
- [ ] under `Additional Data` section there's `uuid`. Check that this `uuid` points to correct encrypted logs.

**Please remove the issue on the Sentry dashboard after finishing tests.**

## Regression Notes
1. Potential unintended areas of impact

Sending contentful issues to `Sentry`.


2. What I did to test those areas of impact (or what existing automated tests I relied on)

I've added unit tests for `WPCrashLoggingDataProvider` and tested this change manually. Also, the logic of the library itself [is unit and manually tested.](https://github.com/Automattic/Automattic-Tracks-Android/tree/develop/AutomatticTracks/src). Finally, the same library is used on `WCAndroid` and the team doesn't report any issues.

3. What automated tests I added (or what prevented me from doing so)

I didn't add any automated tests. I don't see a fit here for them, if you see an opportunity please ping me with an idea.

PR submission checklist:


```
Index: WordPress/src/main/java/org/wordpress/android/util/crashlogging/WPCrashLoggingDataProvider.kt
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- WordPress/src/main/java/org/wordpress/android/util/crashlogging/WPCrashLoggingDataProvider.kt	(revision 4b3b2ec1da9d6236582845d2aba63cb600dc94c2)
+++ WordPress/src/main/java/org/wordpress/android/util/crashlogging/WPCrashLoggingDataProvider.kt	(date 1623673595234)
@@ -37,6 +37,7 @@
     }
 
     override fun crashLoggingEnabled(): Boolean {
+        return true
         if (buildConfig.isDebug()) {
             return false
         }
Index: WordPress/src/main/java/org/wordpress/android/ui/main/MySiteFragment.kt
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- WordPress/src/main/java/org/wordpress/android/ui/main/MySiteFragment.kt	(revision 4b3b2ec1da9d6236582845d2aba63cb600dc94c2)
+++ WordPress/src/main/java/org/wordpress/android/ui/main/MySiteFragment.kt	(date 1623673595246)
@@ -18,6 +18,7 @@
 import androidx.appcompat.widget.TooltipCompat
 import androidx.core.text.HtmlCompat
 import androidx.fragment.app.Fragment
+import com.automattic.android.tracks.crashlogging.CrashLogging
 import com.google.android.material.appbar.AppBarLayout
 import com.google.android.material.snackbar.Snackbar
 import com.wordpress.stories.compose.frame.FrameSaveNotifier.Companion.buildSnackbarErrorMessage
@@ -218,6 +219,7 @@
     @Inject lateinit var unifiedCommentsListFeatureConfig: UnifiedCommentsListFeatureConfig
     @Inject @Named(UI_THREAD) lateinit var uiDispatcher: CoroutineDispatcher
     @Inject @Named(BG_THREAD) lateinit var bgDispatcher: CoroutineDispatcher
+    @Inject lateinit var crashLogging: CrashLogging
     lateinit var uiScope: CoroutineScope
 
     private val selectedSite: SiteModel?
@@ -242,6 +244,7 @@
                     false
             )
         }
+        crashLogging.sendReport(message = "This is just a test. Remove me.")
     }
 
     override fun onDestroy() {
```
